pkg/odo/cli/component/create.go

```diff
@@ -879,6 +873,19 @@ func (co *CreateOptions) devfileRun(cmd *cobra.Command) (err error) {
 		}
 	}
 
+	// set user provided component name in the devfile
+	if co.devfileMetadata.componentName != "" {
+		devObj, err = devfile.ParseFromFile(DevfilePath)
```

Comment:
Why are we not setting the below changes to the devfile before it is written to the disk?

Reply:
Setting these changes before the devfile is written to the disk is not possible with the 2 approaches that we tried - 
1.  We could use `devObj.WriteYamlDevfile()` after modifying the devObj but that is not possible because it does not take the correct path into account.
2. We parse the devfileData to devObj to validate the YAML file, so another option is to convert the devObj back to YAML using the k8s YAML library and write that YAML but that seems to be overridden by the code on [L869](https://github.com/openshift/odo/pull/4927/files#diff-e9cbb4b32525048f3bf6ffce25bb7e24c02221819d5b9de9347fe2c835828058R869).


The second approach seems like a viable option, so I have dropped a comment on https://github.com/openshift/odo/pull/4636#discussion_r680913176 because the code seems redundant to me. If we can remove that code, we can go with the second option.